foxglove_bridge_base/src/serialization.cpp

```diff
@@ -10,14 +10,20 @@ void to_json(nlohmann::json& j, const Channel& c) {
     {"schemaName", c.schemaName},
     {"schema", c.schema},
   };
+
+  if (c.schemaEncoding.has_value()) {
+    j["schemaEncoding"] = c.schemaEncoding.value();
+  }
 }
 void from_json(const nlohmann::json& j, Channel& c) {
-  ChannelWithoutId channelWithoutId{
-    j["topic"].get<std::string>(),
-    j["encoding"].get<std::string>(),
-    j["schemaName"].get<std::string>(),
-    j["schema"].get<std::string>(),
-  };
+  const auto schemaEncoding =
+    j.find("schemaEncoding") == j.end()
+      ? std::optional<std::string>(std::nullopt)
+      : std::optional<std::string>(j["schemaEncoding"].get<std::string>());
```

Comment:
nit
```suggestion
  const std::optional<std::string> schemaEncoding =
    j.find("schemaEncoding") == j.end()
      ? std::nullopt
      : j["schemaEncoding"].get<std::string>();
```

Reply:
Also initially tried that, but CPP is rather strict and complains that the types in the branches (not sure what the correct term is) differs.